clang/lib/Format/WhitespaceManager.cpp

```diff
@@ -14,9 +14,81 @@
 #include "WhitespaceManager.h"
 #include "llvm/ADT/STLExtras.h"
 
+#include <fstream>
+
 namespace clang {
 namespace format {
 
+class Logger {
+public:
+    static Logger& Get() {
+        static Logger instance;
+        return instance;
+    }
+
+    void Init(const std::string &NewFileName) {
+        if (FileName != NewFileName) {
+            LogStream.open(NewFileName.c_str());
```

Comment:
Is there a point in checking `FileName != NewFileName` ? If we try to support re-initialization - it won't work:
"If the stream is already associated with a file (i.e., it is already open), calling this function fails."

If we don't support it - why is this check needed?

Reply:
the purpose was to avoid re-opening the file for every logged line